lib/resources/widgets/text_form_field_for_name.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:flutter/material.dart';
+
+import '../constants/constants.dart';
+import '../global_variables/global_variable.dart';
+
+class TextFormFieldForName extends StatelessWidget {
+  const TextFormFieldForName({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final bodySmall = Theme.of(context).textTheme.bodySmall;
+    final bodyLarge = Theme.of(context).textTheme.bodyLarge;
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(
+          "What is your name?",
+          style: bodyLarge,
+        ),
+        SizedBoxConstants.sizedBoxHeightEight,
+        TextFormField(
+          textInputAction: TextInputAction.next,
+          controller: nameController,
+          decoration: const InputDecoration(
+            hintText: "Name*",
+          ),
+          keyboardType: TextInputType.name,
+          style: bodySmall,
+          validator: (value) {
+            if (value != null && (value.isEmpty || value.length < IntConstants.nameMinLength)) {
+              return "\u{24D8} Name should have at-least ${IntConstants.nameMinLength} characters";
```

Comment:
The Unicode for the alert circle(circle with an exclamation mark inside it) was not found. Tried searching for the Unicode on the internet but was not found.

Reply:
Can you suggest an alternative? What if we had a use case when we had to show a material icon in the error message?

Reply:
The alternative is to use a row widget that has an icon with the error text.

Reply:
@Sujey-Naik This answer is not acceptable. This lacks context. Please explain in detail.

Reply:
There is textEditingController for the text form field of name(nameController). addListener function is added to nameController. I did add a validation function as the parameter for addListener(This method invokes the parameter function every time there is a change in the nameController). There is a boolean value called isNameValid which is initially true. If isNameValid is true, then the Row widget with an error icon and error text is not displayed. If it is false, then the row widget is displayed. In the validation function based on criteria for validation, I change the value of isNameValid to true or false. So whenever there is interaction with the text form field of the name, the validation is done and the error message is shown based on isNameValid boolean value. 
When we submit the form, I first check if the text form field is empty(when user has made no interaction with it). If text form field is empty, I change the value of isNameValid to false and call setState() method to show the error message. Then I check whether the isNameValid and isDobValid is true. Only if it is true then the form will be submitted. Else, the form will not be submitted.
So the error message is controlled using a boolean value.